packages/color-area/src/ColorArea.ts

```diff
@@ -126,57 +126,51 @@ export class ColorArea extends SpectrumElement {
     @property({ attribute: false })
     private activeAxis = 'x';
 
+    private _x = 1;
+
     @property({ type: Number })
     public get x(): number {
         return this.colorController.color.hsv.s / 100;
     }
 
     public set x(x: number) {
-        if (x === this.x) {
+        const oldValue = this._x;
+        // Restrict x to the nearest step
+        const newValue = Math.round(x / this.step) * this.step;
+        if (newValue === oldValue) {
             return;
         }
-        const oldValue = this.x;
-        if (this.inputX) {
-            // Use the native `input[type='range']` control to validate this value after `firstUpdate`
-            this.inputX.value = x.toString();
-            this.colorController.color.set(
-                's',
-                this.inputX.valueAsNumber * 100
-            );
-        } else {
-            this.colorController.color.set('s', x * 100);
-        }
+        this._x = newValue;
+        this.colorController.color.set('s', newValue * 100);
         this.requestUpdate('x', oldValue);
     }
 
+    private _y = 1;
```

Comment:
I see the public getter returns values derived from this.colorController.color.hsv. 
If this.colorController.color.set() is called elsewhere, the cache (_x/_y) and the color model may become inconsistent. You may have two sources of truth for the same value.

Instead consider always deriving x/y from the color model and using the setters only to mutate the model. You may not need _x and _y at all